### PR TITLE
Add API documentation

### PR DESCRIPTION
**Pull request by Widcket**
- [X] All new/changed/fixed functionality is covered by tests (or N/A)
- [X] I have added documentation for all new/changed functionality (or N/A)

### 📋 Changes

This PR adds the API documentation for the new types and methods, and updates the docs for some existing ones.

### CODE REVIEW

File: Auth0/WebAuthUserAgent.swift [hunk 1]
```diff
@@ -1,7 +1,16 @@
 #if WEB_AUTH_PLATFORM
+/// Represents the external user agent used to perform q web-based operation.
```
Comment (stevehobbsdev) [inline Auth0/WebAuthUserAgent.swift:2]: Typo here? "q web-based"..

Comment (Widcket) [inline Auth0/WebAuthUserAgent.swift:2]:
```suggestion
/// Represents the external user agent used to perform a web-based operation.
```

Comment (Widcket) [reply Auth0/WebAuthUserAgent.swift]: Good catch, fixed in https://github.com/auth0/Auth0.swift/pull/697/commits/25bb8faf17450b2b2ff52730b2fac1cfd3485cd8.

File: Auth0/WebAuthUserAgent.swift [hunk 2]
```diff
@@ -1,7 +1,16 @@
...
+/// Represents the external user agent used to perform a web-based operation.
+///
+/// - See: [Example](https://github.com/auth0/Auth0.swift/blob/master/Auth0/SafariProvider.swift)
 public protocol WebAuthUserAgent {
 
+    /// Starts the external user agent.
     func start()
+
+    /// Tears down the external user agent after the web-based operation completed, if needed.
+    /// Auth0.swift will call this method after the callback was received and processed, or after the user cancelled the operation, or after any other error occurred.
```
Comment (Widcket) [inline Auth0/WebAuthUserAgent.swift:11]:
```suggestion
    /// Auth0.swift will call this method after the callback URL was received and processed, or after the user cancelled the operation, or after any other error occurred.
```

**Review by stevehobbsdev — APPROVED** [review]

**Pull request merged**